### DISCUSSION TITLE
NPCBots.patch

### DISCUSSION THREAD

**Discussion by BurnsFast** [trickerer/Trinity-Bots#210] [Q&A]
Apply the patch using patch -p1 < NPCBots.patch command (creating new files)

where is this being applied? and how

i have compiled the server everything works and runs. but what exacly do i need to do to get this into my build.

**Comment by trickerer** [comment]
2. Download NPCBots.patch and put it **into your TrinityCore folder**
3. Apply the patch using `patch -p1 < NPCBots.patch` command (creating new files)

The command has to be executed in your TrinityCore root folder, this is exactly where the patch must be put first.
Use whatever shell you have. If you are on Windows you typically can open `Git Bash` from right-click menu anywhere.
And don't forget to re-build.

**Reply by BurnsFast** [reply]
thank you so much for the very very fast reply... i have been trying to figure this out for well over an hour.... turns out the git i had was corrupted :D but that worked perfectly ....

i would assume after i have done patch -p1 < NPCBots.patch i can delete that file and then rebuild

**Reply by Rylovnikov** [reply]
> 2. Download NPCBots.patch and put it **into your TrinityCore folder**
> 3. Apply the patch using `patch -p1 < NPCBots.patch` command (creating new files)
> 
> The command has to be executed in your TrinityCore root folder, this is exactly where the patch must be put first. Use whatever shell you have. If you are on Windows you typically can open `Git Bash` from right-click menu anywhere. And don't forget to re-build.

........................
Hello, there was a problem with the patch.
I go to the directory where NPCBots.patch lies
cd /home/TrinityCore

I write -
patch -p1 < NPCBots.patch

and in response
patch: **** Only garbage was found in the patch input.
.................
Please help me solve the problem

**Reply by trickerer** [reply]
@Rylovnikov try with `git apply NPCBots.patch`

**Reply by Rylovnikov** [reply]
> @Rylovnikov try with `git apply NPCBots.patch`

Thank you, it worked!
